### PR TITLE
Add `image-suffix` and `image-tag` to keep old `version` behavior

### PR DESCRIPTION
**Pull request by aanm**

### CODE REVIEW

File: install/helm.go [hunk 1]
```diff
@@ -105,6 +105,46 @@ func (k *K8sInstaller) generateManifests(ctx context.Context) error {
 	// older than 1.11.0 we will fix it afterwards.
 	case versioncheck.MustCompile(">=1.9.0")(ciliumVer):
 		// case versioncheck.MustCompile(">=1.11.0")(ciliumVer):
+		// If the user has specified a version with `--image-tag` then
+		// set all image tags with that version. The user will have the
+		// option to overwrite any of these options with the specific
+		// helm option.
+		imageTag := k.params.ImageTag
+		if imageTag != "" {
```
Comment (jrajahalme) [inline install/helm.go:113]: This seems wrong? should test as `if imageTag == ""` ??

File: install/install.go [hunk 2]
```diff
@@ -313,6 +321,17 @@ func (k *K8sInstaller) getCiliumVersion() semver.Version {
 	return v
 }
 
+func (k *K8sInstaller) getImagesSHA() string {
+	ersion := strings.TrimPrefix(k.params.Version, "v")
+	_, err := versioncheck.Version(ersion)
+	// If we got an error then it means this is a commit SHA that the user
+	// wants to install on all images.
+	if err != nil {
```
Comment (jrajahalme) [inline install/install.go:329]: Hmm.. Does this same check also pass "latest" as a "SHA" as it does not parse as a version?

Comment (jrajahalme) [reply install/install.go:329]: If so then we'd not need image-tag option at all?

**Review by jrajahalme — CHANGES_REQUESTED** [review]

File: install/install.go (hunk 2 above)
Comment (aanm) [reply install/install.go:329]: Let's keep the image-tag to have better semantics

**Review by jrajahalme — APPROVED** [review]

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Pull request merged**